model/SyncLog/Storage/RdsSyncLogStorage.php

```diff
@@ -0,0 +1,253 @@
+         }
+         $this->persistenceId = $this->getOption(self::OPTION_PERSISTENCE_ID);
+     }
+
+     /**
+      * @return common_persistence_SqlPersistence
+      */
+     public function getPersistence()
+     {
+         if ($this->persistence === null) {
+             $this->persistence = $this->getServiceLocator()
+                 ->get(\common_persistence_Manager::SERVICE_ID)
+                 ->getPersistenceById($this->persistenceId);
+         }
+
+         return $this->persistence;
+     }
+
+     /**
+      * @return \Doctrine\DBAL\Query\QueryBuilder
+      * @throws \oat\oatbox\service\exception\InvalidServiceManagerException
+      */
+     private function getQueryBuilder()
+     {
+         return $this->getPersistence()->getPlatForm()->getQueryBuilder()->from(self::TABLE_NAME);
+     }
+
+     /**
+      * Store SyncLogEntity in rds storage.
+      *
+      * @param SyncLogEntity $entity
+      * @return mixed|void
+      */
+     public function create(SyncLogEntity $entity)
+     {
+         return $this->getPersistence()->insert(
+             self::TABLE_NAME,
+             [
+                 SyncLogStorageInterface::COLUMN_SYNC_ID => $entity->getSyncId(),
+                 SyncLogStorageInterface::COLUMN_BOX_ID => $entity->getBoxId(),
+                 SyncLogStorageInterface::COLUMN_ORGANIZATION_ID => $entity->getOrganizationId(),
+                 SyncLogStorageInterface::COLUMN_DATA => json_encode($entity->getData()),
+                 SyncLogStorageInterface::COLUMN_STATUS => $entity->getStatus(),
+                 SyncLogStorageInterface::COLUMN_REPORT => json_encode($entity->getReport()),
+                 SyncLogStorageInterface::COLUMN_STARTED_AT => $entity->getStartTime()->format(SyncLogEntity::DATE_TIME_FORMAT)
+             ]
+         );
+     }
+
+     /**
+      * Update synchronization log record.
+      *
+      * @param SyncLogEntity $entity
+      * @return mixed
+      */
+     public function update(SyncLogEntity $entity)
+     {
+         $qb = $this->getQueryBuilder();
+         $qb->update(self::TABLE_NAME)
+             ->set(SyncLogStorageInterface::COLUMN_STATUS, $qb->createNamedParameter($entity->getStatus()))
+             ->set(SyncLogStorageInterface::COLUMN_DATA, $qb->createNamedParameter(json_encode($entity->getData())))
+             ->set(SyncLogStorageInterface::COLUMN_REPORT, $qb->createNamedParameter(json_encode($entity->getReport())));
+
+         $finishTime = $entity->getFinishTime();
+         if ($finishTime instanceof \DateTime) {
+             $finishedAt = $entity->getFinishTime()->format(SyncLogEntity::DATE_TIME_FORMAT);
+             $qb->set(SyncLogStorageInterface::COLUMN_FINISHED_AT, $qb->createNamedParameter($finishedAt));
+         }
+
+         $qb->where(SyncLogStorageInterface::COLUMN_ID . ' = ' . $qb->createNamedParameter($entity->getId()));
+
+         return $qb->execute();
+     }
+
+     /**
+      * @param $id
+      * @return SyncLogEntity
```

Comment:
you returns an array

Reply:
but I think this is a good idea to return an entity

Reply:
Entity is created in service which calls this method (you will see it in PR #86)